data/scripts/move_tutors.inc

```diff
@@ -266,5 +266,6 @@ MoveTutor_EventScript_OpenPartyMenu::
 	return
 
 MoveTutor_EventScript_CanOnlyBeLearnedOnce::
+	special DisableMsgBoxWalkaway
```

Comment:
I understand _why_ you did this and I don't think your logic is wrong, but it IS different than how we tend to approach additions like this.